controller/src/statemachine.rs

```diff
@@ -47,3 +71,47 @@ pub fn determine_next_node_spec(brs: &BottlerocketShadow) -> BottlerocketShadowS
         }
     }
 }
+
+fn is_time_reached(node_status: &BottlerocketShadowStatus) -> bool {
```

Comment:
This name is a bit too vague. I might suggest something like this with a docstring:

```suggestion
/// Returns whether or not an Idle node is allowed to enter an update workflow.
/// This returns false if the node has previously encountered an error and not yet
/// passed its retry timer.
fn node_allowed_to_update(node_status: &BottlerocketShadowStatus) -> bool {
```